include/swift/SIL/SILBuilder.h

```diff
@@ -980,6 +980,8 @@ class SILBuilder {
   RetainValueInst *createRetainValue(SILLocation Loc, SILValue operand,
                                      Atomicity atomicity) {
     assert(isParsing || !getFunction().hasQualifiedOwnership());
+    assert(!SILModuleConventions(getModule()).useLoweredAddresses() ||
```

Comment:
This disjunction appears in many places in SIL and SILGen. Can we factor it out into a new method on SILType?

Reply:
@atrick what do you think?
This is part of Andy's work on semantic SIL. Makes sense to me that it is a module's calling convention / an address-only type can be used as a value before lowering with that enabled.

Reply:
In the general case, we want a single idiom `SILModuleConventions(getModule()).useLoweredAddresses()` to search for and remove once that mode is no longer supported in whatever SIL pass is using it.

In this case, it's in the SILBuilder itself, and we probably still want to use the SILBuilder after address lowering. So, in this case Slava's suggestion makes sense... but only in the SILBuilder. Not sure what to call it. `type.isLoadableOrLowered()`?

Reply:
Done. added `type.isLoadableOrLowered` and used it in the builder